draft-dnsop-deleg-transport.md

```diff
@@ -100,76 +100,82 @@ contributor:
 
-Legacy DNS resolvers unaware of DELEG mechanism would continue to use the NS and DS records, while resolvers that understand DELEG and its associated parameters can efficiently switch to new transports.
+Legacy DNS resolvers that are unaware of the DELEG mechanism would continue to use the NS and DS records, while resolvers that understand DELEG and its associated parameters can efficiently switch to new transports.
 
 ## Terminology
 
 The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT",
 "SHOULD", "SHOULD NOT", "RECOMMENDED", "NOT RECOMMENDED", "MAY", and
-"OPTIONAL" in this document are to be interpreted as described in \\
+"OPTIONAL" in this document are to be interpreted as described in
 BCP 14 {{?RFC2119}} {{?RFC8174}} when, and only when, they appear in
 all capitals, as shown here.
 
 Terminology regarding the Domain Name System comes from {{?BCP219}}, with addition terms defined here:
 
-* [tbd]
+* transport parameters	  The set of attributes that define how to reach the nameserver specified in a ServiceMode DELEG record.
 * [tbd]
 
 ## Introductory Examples
 
-To introduce DELEG record with DNS-over-TLS {{?RFC7858}}, this example shows a possible response from an authoritative in the authority section of the DNS response when delegating to another nameserver.
-
+To use a DELEG record to point a resolver to an authority that supports DNS-over-TLS {{?RFC7858}}, this example shows a possible response from an authoritative server in the Authority section of the DNS response:
 
-    example.com.  86400  IN DELEG  1 ns2.example.com. (
+    example.com. 86400  IN DELEG 1 ns2.example.com. (
                     alpn=dot
                     tlsa="2 0 1 ABC..."
                     ipv4hint=192.0.2.54
                     ipv6hint=2001:db8:2423::3 )
-    example.com.  86400  IN NS     ns1.example.com.
-    ns1.example.com.    86400   IN  A  192.0.2.1
-    ns1.example.com     86400   IN  AAAA    2001:DB8::1
+    example.com. 86400  IN NS ns1.example.com.
 
-In this example, the authoritative nameserver is delegating using DNS-over-TLS.
-Like in SVCB, DELEG also offer the ability to use the AliasMode delegation. The example below shows an example where example.com is being delegated with a DELEG AliasMode record which can then be further resolved using standard SVCB to locate the actual parameters.
+The alpn attribute specifies DNS-over-TLS, with the tlsa attribute
+providing the necessary cryptographic material to establish the
+connection.  The ipv4hint and ipv6hint serves the traditional role of
+glue, since the ns2 server lives in the example.com zone.
 
-    example.com.  86400  IN DELEG 0   config2.example.net.
-    example.com.  86400  IN NS     ns1.example.net.
+The Additional section would continue to serve glue records for the
```

Comment:
This is intriguing but I don't quite believe it.  DELEG is in the parent zone, and is only retrieved by an explicit query (QTYPE=DELEG).  Sure, the parent should serve glue records alongside an NS record Answer, but why should it serve glue records alongside a DELEG Answer?

There's another wrinkle here though: SVCB recommends putting A+AAAA records for TargetName in the Additional section if they are in the same zone, to avoid a followup query.  If we apply that to DELEG, it could mean adding A+AAAA records for "ns2.example.com" (in the above example).  However, this would be redundant with the IP hints, and it would also be logically different: those A+AAAA records are locally available but they _aren't_ part of this zone.

Bottom line: we should do IP glue or IP hints, but not both.  Overall, my preference would be to stick with hints and get rid of the whole concept of glue.

Reply:
> is only retrieved by an explicit query (QTYPE=DELEG)

That's not really a settled matter.  I still think, pending testing to confirm that for the vast majority of the resolver base, that the real deployment win here is to serve NS, DS and DELEG together for any delegation and NOT require an explicit query for them.  We're trying to make the whole business of finding modern auths more efficient, not add more queries. 

And I really, really don't think we should be adding addresses for the DELEG names to the Additional section, because something that understands what DELEG is at all should be using the hints.

Reply:
It seems like this is not settled, and is also not relevant to the transport draft.  Please remove it or move it to the other draft.

BTW, using QTYPE=DELEG doesn't add queries.  It just replaces the "arbitrary" QTYPE in QNAME minimization with a meaningful one.  Non-DELEG servers _should_ treat it as an unknown QTYPE and return the NS (and glue) in a delegation response.